exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/CsrfTokenInjectFilter.java

```diff
@@ -43,11 +43,14 @@ public void init(FilterConfig filterConfig) throws ServletException {
   public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
     HttpServletRequest httpRequest = (HttpServletRequest) request;
     if (HttpMethod.GET.equalsIgnoreCase(httpRequest.getMethod())) {
+      // We don't create a session with this call as we need to check if there is a session already (i.e. if a user is logged in).
       HttpSession session = httpRequest.getSession(false);
       if (session != null) {
         String csrfToken = (String) session.getAttribute(WebServerConstants.CSRF_TOKEN);
         if (csrfToken == null) {
-          csrfToken = RandomStringUtils.random(20, 0, 0, true, true, null, new SecureRandom());
+          byte[] buffer = new byte[32];
```

Comment:
As I have asked before please explain choice of creating the token. Even better extract it into separate method and add good java doc.

Reply:
Please don't forget to extract token creation into separate method...

Reply:
Done.